clippy_lints/src/utils/internal_lints.rs

```diff
@@ -207,3 +218,28 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for LintCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.cx.tcx.hir)
     }
 }
+
+
+
+pub struct PreferFxHash;
+
+impl LintPass for PreferFxHash {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(PREFER_FX_HASH)
+    }
+}
+
+impl EarlyLintPass for PreferFxHash {
+    fn check_ident(&mut self, cx: &EarlyContext<'_>, ident: Ident) {
+        if let Some(msg) = {
```

Comment:
You can get rid of the `if let` and the `Some(_)` if you use an early `return` in the `else` case.

Reply:
Oh neat. Will do.